bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/dialogs/RecentNewWizardSelection.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2023 ETAS GmbH and others, all rights reserved.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ETAS GmbH - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.ui.internal.dialogs;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+/**
+ * A <code>RecentNewWizardSelection</code> is used to manage the five most
+ * recent new wizards used or created from the "Other" shortcut, which is part
+ * of the menu manager with New Wizard actions. The recently used new wizards
+ * will appear before the "Other" shortcut.
+ *
+ * @since 3.5
+ */
+public class RecentNewWizardSelection {
+	private Set<String> selectedFromOther = new LinkedHashSet<>();
+	private static RecentNewWizardSelection instance;
+	private static final int MAX_MENU_SIZE = 5;
```

Comment:
If you intended to have 5 recent items: currently only **four** are kept.

Reply:
> If you intended to have 5 recent items: currently only **four** are kept.

Could you please share the scenario you tested .
 It do not add the shortcuts if it is already present as part of fixed shortcuts (highlighted in the image) of current perspective . The last used shortcuts change if you switch the perspective .
 
![image](https://github.com/user-attachments/assets/ec803ebd-3132-40b8-98f3-9345b866c880)